src/main/kotlin/org/malachite/estella/process/domain/RecruitmentProcessDto.kt

```diff
@@ -30,15 +36,37 @@ fun RecruitmentProcess.toRecruitmentProcessDto() = RecruitmentProcessDto(
 
 data class TaskDto(
     val id: Int?,
-    val tests: String,
-    val description: String,
+    val testsBase64: String,
+    val descriptionFileName: String,
+    val descriptionBase64: String,
     val timeLimit: Int,
     val deadline: Timestamp
 )
+
+fun TaskDto.toTask(): Task {
+    print(String(Base64.getDecoder().decode(this.descriptionBase64)))
+    return Task(
+        this.id,
+        SerialBlob(Base64.getDecoder().decode(this.testsBase64)),
+        SerialClob(String(Base64.getDecoder().decode(this.descriptionBase64)).toCharArray()),
+        this.descriptionFileName,
+        this.timeLimit,
+        this.deadline
+    )
+}
+
 fun Task.toTaskDto() = TaskDto(
     id,
-    tests.toString(),
-    description.toString(),
+    Base64.getEncoder().encodeToString(this.tests.getBytes(1, this.tests.length().toInt())),
```

Comment:
Czy to jest jakiś oczywisty kawałek kodu czy też tu się dzieje jakiś magic?

Reply:
Magic i to taki całkowity, ale robimy coś podobnego przy job seeker files, a tam to działa XDD